src/Hazelcast.Net.Tests/Serialization/Compact/CompactQaTests.cs

```diff
@@ -99,46 +93,31 @@ private async Task ExceptionPreventsClientFromReconnectingTask(bool recover)
         await RcClient.StopMemberWaitRemovedAsync(client, RcCluster, member).CfAwait();
 
         // trigger exceptions
-        throwException = true;
-        if (recover)
-        {
-            async Task RecoverAfter(int delayMilliseconds)
-            {
-                await Task.Delay(delayMilliseconds);
-                throwException = false;
-            }
+        Volatile.Write(ref throwException, true);
+        await conQueue.SuspendAsync();
+
 
-            var recoverTask = RecoverAfter(5000); // fire-and-forget
+        async Task RecoverAfter(int delayMilliseconds)
+        {
+            await Task.Delay(delayMilliseconds);
+            Volatile.Write(ref throwException, false);
+            conQueue.Resume();
         }
 
+        var recoverTask = RecoverAfter(5000); // fire-and-forget
+
+
         HConsole.WriteLine(this, "-------- START MEMBER --------");
 
         // start another member
         member = await RcClient.StartMemberAsync(RcCluster).CfAwait();
 
         HConsole.WriteLine(this, "-------- PUT ASYNC --------");
 
-        if (recover)
```

Comment:
why removed? can you explain in detail the change in this file?

Reply:
This test cannot accomplish on targeted testing. It is bended to much but we can't test the client with that approach. So, I've refactored that part.